coordinator/coordinator.go

```diff
@@ -276,24 +270,32 @@ func (c *Coordinator) WatchForNewJobs() error {
 	c.Logger.Info("Watching for new jobs")
 	// TODO: change it so we can watch for new jobs using task manager; searching for PENDING status tasks
 	// JOB__TYPE__NAME__VARIANT
-	runs, err := c.taskManager.GetAllTaskRuns()
+	runs, err := c.Metadata.Tasks.GetAllRuns()
 	if err != nil {
 		return fferr.NewInternalError(err)
 	}
-	runs.FilterByStatus(scheduling.Pending)
+
+	runs.FilterByStatus(scheduling.PENDING)
 
 	for _, run := range runs {
 		go func(run scheduling.TaskRunMetadata) {
-			// LOCK and pass the key to the worker; if lock fails then skip
-			// UNLOCK after the worker is done
-			// LockTaskRun(taskID TaskID, runId TaskRunID) (sp.LockObject, error)
-			lock, err := c.taskManager.LockTaskRun(run.TaskId, run.ID)
+			lock, err := c.Locker.LockRun(run.ID)
```

Comment:
For memory, this should work fine because this map will be separate from the Task Map. But for etcd, we will run into an issue where this locks it then Task can't lock it. If I am thinking correctly.

Reply:
if ahmad is right this needs to be documented. Could be v confusing to work with

Reply:
This is a separate locker im using. It wont interfere with the locking implementation